### ISSUE TITLE
Save builds | user folder with japanese symbols

### ISSUE THREAD

**Issue by t-h-e-r-e-d-p-i-l-l** [LocalIdentity/PathOfBuilding#3748]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

When trying saving build.

### What is the actual behaviour/value?

Error while saving.

### Build code

```shell
https://pastebin.com/2bC2hdu4
```

### Screenshots

![image](https://user-images.githubusercontent.com/65071735/140761939-82e74411-ec01-4964-826a-3441fa2e8b8d.png)

**Comment by zao** [comment]
Path of Building currently uses narrow (A) Windows APIs for things like file access and paths where it's very reliant on that paths used can be encoded in the user's ANSI codepage (typically Windows-1252 or Shift-JIS (CP932)).

I have some questions:

Are you using a Japanese locale on your system and do you know if the username can be represented cleanly in that codepage?

Have you been able to use PoB with this username in the past?

Are you using some software that could impact how files in your Documents folder are accessed like OneDrive?

Can you see any existing builds in the build list in that directory?

If it is not sensitive, would you mind sharing the username?

**Comment by TAGRUAT0** [comment]
Windows 11 (FULL ENG)
![image](https://user-images.githubusercontent.com/65071735/141376431-cd70906f-a89e-4e8d-8eb9-5e688087cb22.png)
OneDrive fully removed

**Comment by zao** [comment]
Which "Language for non-Unicode programs" are you using on this Windows installation?
You can see it under the Region control panel:
![image](https://user-images.githubusercontent.com/260218/141469986-743e0c73-8ce2-4337-aef2-3a792f3c688b.png)

You can also tell which codepage is in use by running `chcp` in the command prompt. 437 is the US one, 932 is the Japanese one.

In general usernames with characters that cannot be represented in the system locale is a major compatibility hazard and it's quite likely that other software also may malfunction.

Until we rewrite several parts of the Path of Building runtime to further internationalize it, I recommend reinstalling to a directory with a friendlier path, and also move your build directory to a similar location.

**Comment by TAGRUAT0** [comment]
![image](https://user-images.githubusercontent.com/65071735/141478007-6ee45d02-da26-4919-9432-491c79a34307.png)

**Comment by zao** [comment]
Yeah, this is definitely outside of the capability of the current Path of Building runtime.

**Comment by TAGRUAT0** [comment]
Sorry, i did not know about it
![image](https://user-images.githubusercontent.com/65071735/141517399-22659e8e-98bb-462a-9094-7123714931db.png)
How temporary solution is it good.

**Comment by zao** [comment]
Try it, it might work OK.
You may have to install PoB in a different place that doesn't include this user name for full functionality.

**Comment by TAGRUAT0** [comment]
"Build save path" Works fine.
